Fix private package install docs (#1846)
astronomer/docs: astro/develop-project.md

#### Step 1: Specify the private repository in your project

> To add a Python package from a private repository to your Astro project, specify the repository’s SSH URL in a new `private-requirements.txt` file. The URL should use the following format:

Reviewer: ```suggestion
To add a Python package from a private repository to your Astro project, specify the Secure Shell (SSH) URL for the repository in a new `private-requirements.txt` file. Use the following format for the SSH URL:
```

1. Optional. Copy and save any existing build steps in your `Dockerfile`.

2. Add `openssh-client` and `git` to your `packages.txt` file.

3. In your Dockerfile, add the following instructions:

>     ```docker

Author: I removed this as I felt it was implied, but happy to add it back:
```suggestion
    ```docker
    FROM quay.io/astronomer/astro-runtime:7.2.0
```

    RUN mkdir -p -m 0700 ~/.ssh && \
        echo "github.com ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIOMqqnkVzrm0SdG6UOoqKLsabgH5C9okWi0dh2l9GKJl" >> ~/.ssh/known_hosts

    COPY private-requirements.txt .
    RUN --mount=type=ssh,id=github pip install --no-cache-dir --requirement private-requirements.txt
    ENV PATH="/home/astro/.local/bin:$PATH"
    ```

    In order, these instructions:

>     - Adds the fingerprint for GitHub to known_hosts
>     - Copies you private-requirements.txt file into the image
>     - Install Python-level packages from your private repository as specified in your `private-requirements.txt` file. Note, this securely mounts your SSH key at build time. This ensures that the key itself is not stored in the resulting Docker image filesystem or metadata.
>     - Adds the user bin directory onto PATH

Reviewer 2: ```suggestion
    - Add the fingerprint for GitHub to `known_hosts`
    - Copy your `private-requirements.txt` file into the image
    - Install Python-level packages from your private repository as specified in your `private-requirements.txt` file. Tthis securely mounts your SSH key at build time, ensuring that the key itself is not stored in the resulting Docker image filesystem or metadata.
    - Add the user bin directory onto `PATH`
```

  :::info

>   If your repository is not hosted on GitHub, replace the fingerprint with one from where the package is hosted. `ssh-keyscan` can be used to generate the fingerprint, if you don't already know it.

Reviewer: ```suggestion
  If your repository isn't hosted on GitHub, replace the fingerprint with the hosted package location. Use `ssh-keyscan` to generate the fingerprint,
```
